std/evmprecompiles/08-bnpairing.go

```diff
@@ -0,0 +1,21 @@
+package evmprecompiles
+
+import (
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/algebra/emulated/sw_bn254"
+)
+
+// ECPair implements [ALT_BN128_PAIRING_CHECK] precompile contract at address 0x08.
+//
+// [ALT_BN128_PAIRING_CHECK]: https://ethereum.github.io/execution-specs/autoapi/ethereum/paris/vm/precompiled_contracts/alt_bn128/index.html#alt-bn128-pairing-check
+func ECPair(api frontend.API, P *sw_bn254.G1Affine, Q *sw_bn254.G2Affine) *sw_bn254.GTEl {
+	pair, err := sw_bn254.NewPairing(api)
+	if err != nil {
+		panic(err)
+	}
+	R, err := pair.Pair([]*sw_bn254.G1Affine{P}, []*sw_bn254.G2Affine{Q})
```

Comment:
I think the precompile does not return the multi-pairing value `R` but rather checks if `e(P,Q)==1`. So maybe instead we should use 
https://github.com/ConsenSys/gnark/blob/6939cf5ba04cd000024b7120a69ea9cfcf9b2f6d/std/algebra/emulated/sw_bn254/pairing.go#L219

Reply:
Good catch! Changed. Also changed the corresponding test.